### PR TITLE
k_timer tracing with CTF, Systemview

### PR DESCRIPTION
**Pull request by nashif**
- tracing: ctf: remove unused metadata
- tracing: ctf: add timer support

### CODE REVIEW

**Review by teburd — DISMISSED** [review]

**Comment by tleksell-pe** [comment]
@nashif This change makes sense, duration and period should be part of the trace macro. Unfortunately, this change breaks the Percepio trace module since it expands the parameters for sys_port_track_k_timer_start.

While I could fix this specific macro for our module, I'm thinking that it might be better to improve our modules resilience to macro updates (new parameters) by changing our macro mappings from sys_port_hook(x, y) to sys_port_hook(x, y, ...). I'll get started on an update (I should have it ready for merge early next week) so we can get this change merged ASAP.

Looking through various trace hooks I've noticed several others that should be updated similar to this one (obvious parameters missing in the hook). I'll see about updating these in Zephyr soon.

**Comment by nashif** [comment]
> @nashif This change makes sense, duration and period should be part of the trace macro. Unfortunately, this change breaks the Percepio trace module since it expands the parameters for sys_port_track_k_timer_start.

yes, I realized this will break percepio, will not merge until we have a fix on your side.

File: subsys/tracing/ctf/ctf_top.c [hunk 1]
```diff
@@ -274,24 +274,48 @@ void sys_trace_k_mutex_unlock_exit(struct k_mutex *mutex, int ret)
...
+	ctf_top_timer_init(
+		(uint32_t)(uintptr_t)timer);
 }
 
 void sys_trace_k_timer_start(struct k_timer *timer, k_timeout_t duration,
 			     k_timeout_t period)
 {
+	ctf_top_timer_start(
+		(uint32_t)(uintptr_t)timer,
+		k_ticks_to_ms_floor32((uint32_t)duration.ticks),
```
Comment (teburd) [inline subsys/tracing/ctf/ctf_top.c:288]: I don't think we should be rounding the ticks off to ms, us perhaps, but not ms.

**Review by teburd — CHANGES_REQUESTED** [review]
I believe chopping off the ticks to ms will result in confusing values for duration/period in the trace.

**Review by teburd — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**